go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -98,17 +101,19 @@ func TestComputeFilesWithXattr(t *testing.T) {
 				t.Fatalf("Failed to create tmp file for testing digests: %v", err)
 			}
 			after := time.Now().Truncate(time.Second).Add(time.Second)
-			defer os.RemoveAll(filename)
+			t.Cleanup(func() { os.RemoveAll(filename) })
 			got := Compute(filename)
+			wantDigest, wantErr := digest.NewFromString(fmt.Sprintf("%s/%d", tc.name, len(tc.contents)))
```

Comment:
Isn't an err here a failure in the setup of the test? We should never except an error here?

Reply:
The reason why I have this wantErr here is because the current `getXAttrMock()` in this test will try to return filename itself as the digest_hash, which is less than 64 bytes, and it actually raise an error after I changed the filemetadata.go code.

I got the point you mentioned here, so I have changed the code for this test so that `getXAttrMock()` will just return a valid 64 bytes long const hash string (we will only check happy path here for this existing test).

The error path which will raise an error when generating digest for metadata is covered by the new test I added below.

I think the testing logic is more clear now, with minimum change to our existing code.